testing/python/metafunc.py

```diff
@@ -1503,6 +1548,31 @@ def test_it(x): pass
         result = pytester.runpytest()
         assert result.ret == 0
 
+    def test_parametrize_module_level_test_with_class_scope(
```

Comment:
Thanks, I think this test is good now.

I suggest adding a docstring:

Test that a class-scoped parametrization without a corresponding `Class` gets module scope, i.e. we only create a single FixtureDef for it per module.

Reply:
Shall I add a `,however its parametrization scope remains the same`? Since we keep the same scope for `FixtureDef`'s scope arg in both `add_metafunc_pseudo_fixture_def` and the new `metafunc.parametrize`.

Reply:
I'm pretty sure that the `fixturedef.scope` needs to be `class` in this case, for things like the "can't request lower scope from higher scope" check, introspection, and probably other stuff. It seems safer to me to keep the original scope and do the "find the *actual* scope" dance only where needed.

Reply:
I mean if we swap `test_1` and `test_2`, the scope attribute of `x` fixturedef would be `module`. Is this OK?

Reply:
However, It does not affect reordering since we use `callspec._arg2scope` for creating fixture keys.

Reply:
> I mean if we swap test_1 and test_2, the scope attribute of x fixturedef would be module as test_2 is collected earlier then. Is this OK?

It's dubious, I'd say it's a bug (preexisting of course).